src/java/org/apache/cassandra/auth/IInternodeAuthenticator.java

```diff
@@ -35,10 +36,53 @@
      */
     boolean authenticate(InetAddress remoteAddress, int remotePort);
```

Comment:
We can but I think this method is valid for IP based authenticator.

Reply:
I think it would be good to mark it as deprecated. Implementations should move towards `authenticate(InetAddress remoteAddress, int remotePort,
                                  Certificate[] certificates, InternodeConnectionType connectionType)` and just ignoring the certificate/connectionType if not needed.

Reply:
sounds good!